### PR TITLE
feat: add landing page and improve header

### PR DESCRIPTION
**Pull request by jiejasonliu**
### New Features
- new responsive landing page skeleton
- improved 404 page
- `echoStudy` homepage button now lives in the header
  - also actually works and you can middle mouse for new-tab too
- made header slightly more responsive (search-bar hides, resizing, etc)

---

### Codebase Changes
- introduction of "layouts" to routes; each layout controls how its child routes are rendered
  - currently have two layouts
    - "with sidebar" layout
    - "generic full screen" layout
- started to add media queries for responsiveness 🥳 
- 
---
<details>
<summary><h3>Demo / Showcase (click to open)</h3></summary>

https://user-images.githubusercontent.com/29434693/188813669-6e963aeb-0454-447a-8539-3bb239174cd4.mp4

https://user-images.githubusercontent.com/29434693/188813691-20be7ef8-b96e-48c2-a0c1-bc3c80c38b7b.mp4
</details>

### CODE REVIEW

File: src/routing/paths.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const enum paths {
```
Comment (jiejasonliu) [inline src/routing/paths.ts:1]: fyi, had to move this enum into its own file to break circular dependencies

File: src/layouts/sidebar-layout/sidebar-layout.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Outlet } from 'react-router-dom';
```
Comment (derricklee43) [inline src/layouts/sidebar-layout/sidebar-layout.tsx:2]: Oh, that is dope. So basic all the routes render inside of this wrapper, right?

File: src/pages/landing-page/landing-page.scss [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+      .learn-anything {
+        position: absolute;
+        margin-top: 2vw;
+        right: 8px;
+        text-align: right;
+      }
+    }
+
+    .wave-image {
+      animation: pulse-animation 3s ease-in-out infinite alternate;
```
Comment (derricklee43) [inline src/pages/landing-page/landing-page.scss:114]: I really like this pulse animation. It's pretty dope

**Review by derricklee43 — APPROVED** [review]
lgtm

File: src/layouts/sidebar-layout/sidebar-layout.tsx (hunk 2 above)
Comment (jiejasonliu) [reply src/layouts/sidebar-layout/sidebar-layout.tsx:2]: yup, all child `Route`(s) of this layout

File: src/pages/landing-page/landing-page.scss (hunk 3 above)
Comment (jiejasonliu) [reply src/pages/landing-page/landing-page.scss:114]: yay! I figure it's something we keep until we have a better idea of what we want to do w it

**Pull request merged**